dbt_semantic_interfaces/validations/where_filters.py

```diff
@@ -0,0 +1,268 @@
+from dbt_semantic_interfaces.type_enums import TimeGranularity
+from dbt_semantic_interfaces.validations.validator_helpers import (
+    FileContext,
+    MetricContext,
+    SavedQueryContext,
+    SavedQueryElementType,
+    SemanticManifestValidationRule,
+    ValidationContext,
+    ValidationIssue,
+    ValidationWarning,
+    generate_exception_issue,
+    validate_safely,
+)
+
+
+class WhereFiltersAreParseableRule(SemanticManifestValidationRule[SemanticManifestT], Generic[SemanticManifestT]):
+    """Validates that all Metric WhereFilters are parseable."""
+
+    @staticmethod
+    def _validate_time_granularity_names_impl(
+        location_label_for_errors: str,
+        filter_call_parameter_sets_with_context: Sequence[Tuple[ValidationContext, FilterCallParameterSets]],
+        custom_granularity_names: List[str],
+    ) -> Sequence[ValidationIssue]:
+        issues: List[ValidationIssue] = []
+
+        valid_granularity_names = [
+            standard_granularity.value for standard_granularity in TimeGranularity
+        ] + custom_granularity_names
+
+        for context, parameter_set in filter_call_parameter_sets_with_context:
+            for time_dim_call_parameter_set in parameter_set.time_dimension_call_parameter_sets:
+                if not time_dim_call_parameter_set.time_granularity_name:
+                    continue
+                if time_dim_call_parameter_set.time_granularity_name.lower() not in valid_granularity_names:
+                    issues.append(
+                        ValidationWarning(
+                            context=context,
+                            message=f"Filter for `{location_label_for_errors}` is not valid. "
+                            f"`{time_dim_call_parameter_set.time_granularity_name}` is not a valid granularity name. "
+                            f"Valid granularity options: {valid_granularity_names}",
+                        )
+                    )
+        return issues
+
+    @staticmethod
+    def _validate_time_granularity_names_for_saved_query(
+        saved_query: SavedQuery, custom_granularity_names: List[str]
+    ) -> Sequence[ValidationIssue]:
+        where_param = saved_query.query_params.where
+        if where_param is None:
+            return []
+
+        return WhereFiltersAreParseableRule._validate_time_granularity_names_impl(
+            location_label_for_errors="saved query `{saved_query.name}`",
+            filter_call_parameter_sets_with_context=[
+                (
+                    SavedQueryContext(
+                        file_context=FileContext.from_metadata(metadata=saved_query.metadata),
+                        element_type=SavedQueryElementType.WHERE,
+                        element_value=where_filter.where_sql_template,
+                    ),
+                    where_filter.call_parameter_sets,
+                )
+                for where_filter in where_param.where_filters
```

Comment:
Here, we're looping through the where filters once, then we end up looping through them again in `_validate_time_granularity_names_impl()`. This parameter also seems a bit fragile / easy to mess up.
Instead, can we just validate one `WhereFilter` in `_validate_time_granularity_names_impl()`? Then we would only one loop, at the call site. And that would allow us to take 2 clearly-typed params in the function instead of building `filter_call_parameter_sets_with_context`.

Reply:
> Instead, can we just validate one WhereFilter in _validate_time_granularity_names_impl()?

I *can*.  The fact that I didn't is probably an artifact of originally avoiding changes to the metrics validation logic since it doesn't use `WhereFilter`s directly.

Buuut I am also not sure it's the right path?  metrics validations use `metric.filter.filter_expression_parameter_sets`.  `metrics.filter` is a WhereFilterIntersection, not a WhereFilter, and there's some parsing error logic baked into its `filter_expression_parameter_sets`.  I can change it out pretty easily, but I lack the context to know if that change is likely to upset folks.

------------

However, maybe a compromise that accomplishes the same thing would be to just validate on a single set of `FilterCallParameterSets` per call to `_validate_time_granularity_names_impl`?  I think that's can be done safely with no meaningful change to how this behaves anywhere.

Reply:
Ooo let me look at the metric filters, I didn't realize they had a different type!

Reply:
From discussion: avoid "implementation" as that word is overloaded here.  rename it in next version.